test/release/examples/primers/forallLoops.chpl

```diff
@@ -0,0 +1,305 @@
+
+As with for-loops, the body of a forall statement is a statement or a
+a block statement, whereas the body of a forall expression is an expression.
+
+"Must-parallel" forall statement
+--------------------------------
+
+In the following example, the forall loop iterates over the array indices
+in parallel:
+*/
+
+config const n = 9;
+var A: [1..n] real;
+
+forall i in 1..n {
+  A[i] = i;
+}
+
+writeln("After setting up, A is:");
+writeln(A);
+writeln();
+
+/*
+If ``A`` were a distributed array, each loop iteration would be
+executed on the locale where the corresponding array element resides.
+
+"Must-parallel" forall expression
+---------------------------------
+
+The following forall-expression produces new values in parallel.
+We store these values in a new array.
+*/
+
+var B = forall a in A do a * 3;
+
+writeln("After initialization, B is:");
+writeln(B);
+writeln();
+
+/*
+Zippered "must-parallel" forall statement
+-----------------------------------------
+
+We can iterate over multiple arrays in parallel with a zippered loop.
+Here we illustrate zippering arrays and domains:
+*/
+
+var C: [1..n] real;
+forall (a, b, i) in zip(A, B, C.domain) do
+  C[i] = a * 10 + b / 10;
+
+writeln("After a zippered loop, C is:");
+writeln(C);
+writeln();
+
+/*
+"May-parallel" forall statement
+-------------------------------
+
+The iterator ``onlySerial`` defined below does not have any parallel
+forms. Therefore the may-parallel loop ``[i in onlySerial(n)]``
+will accept it, executing its iterations serially:
+*/
+
+iter onlySerial(m: int) {
+  for j in 1..m do
+    yield j;
+}
+
+[i in onlySerial(n)] {
+  writeln("in onlySerial iteration #", i);
+}
+writeln();
+
+/*
+Using the following must-parallel loop instead would cause
+a "parallel iterator is not found" error:
+
+  .. code-block:: chapel
+
+      forall i in onlySerial(n) {
+        writeln("in iteration #", i);
+      }
+
+"May-parallel" forall expression
+--------------------------------
+
+Given that these arrays provide parallel iteration, the following
+may-parallel expression will be computed in parallel:
+*/
+
+var D = [(a,b,c) in zip(A,B,C)] a + c - b;
+
+writeln("The result of may-parallel expression, D is:");
+writeln(D);
+writeln();
+
+/*
+As with must-parallel loops, if A were a distributed array,
+iteration locality while computing the above expression
+would be determined by A's domain map.
+
+Task Intents and Shadow Variables
+---------------------------------
+
+A forall loop may name some variables declared outside the loop,
+or "outer variables". If so, "shadow variables" are introduced.
+Each task created by the parallel iterator gets its own set of
+shadow variables, one per outer variable.
+
+ - Each shadow variable behaves as if it were a formal argument
+   of the task function for the task. The outer variable is passed
+   to this formal argument according to the argument intent
+   associated with the shadow variable, which is called a "task intent".
+
```

Comment:
The documentation uses the term `forall intent`.

See e.g., [here](https://chapel-lang.org/docs/master/language/spec/data-parallelism.html#forall-intents), [here](https://chapel-lang.org/docs/master/language/spec/task-parallelism-and-synchronization.html#task-intents) and [here](https://chapel-lang.org/docs/master/technotes/reduceIntents.html)

Reply:
@bradcray - Lee is right, the spec is using "forall" intents.
What do you think - let it be a mismatch for now, expecting the audiences for spec vs. primer to be mostly disjoint, revert the primer, (a stretch) switch the spec to "task intents", something else?

Reply:
I wouldn't try to update the spec today, but would ultimately prefer we refer to these as "task variables" everywhere (but think this should be a broader discussion than me just asserting it).  I don't feel strongly about whether we refer to these as task variables or forall variables in this primer at this point.  (I think it's telling that the description is virtually identical to the task intent description, which I think backs my point that we shouldn't pretend they're different things with different terms).